libtransmission/CMakeLists.txt

```
@@ -2,6 +2,16 @@ project(libtr)
 
 configure_file(version.h.in version.h)
 
+set(THIRD_PARTY_SOURCES ConvertUTF.c jsonsl.c wildmat.c)
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  set(DISABLE_WARNINGS -w)
+elseif(CMAKE_C_COMPILER_ID STREQUAL "Clang")
```

On Mac, Apple's Clang is detected as "AppleClang" so it might be better to do
```suggestion
elseif(CMAKE_C_COMPILER_ID MATCHES "Clang")
```